pages/[id].js

```diff
@@ -0,0 +1,57 @@
+import plants from "@/assets/plants.js";
+import { useRouter } from "next/router";
+import Image from "next/image";
+import styled from "styled-components";
+import BackArrow from "@/public/circle-chevron-left.svg";
+import WaterNeed from "@/components/icons/WaterNeed";
+import LightNeed from "@/components/icons/LightNeed";
+import FertiliserSeason from "@/components/icons/FertiliserSeason";
+import Link from "next/link";
+
+export default function PlantDetails() {
+  const router = useRouter();
+  const { id } = router.query;
+
+  if (!id) return <p>Loading...</p>;
+
+  const plant = plants.find((plant) => plant.id === id);
+
```

Comment:
But here you need to do a check:
If the id is not valid (add a non existing id to your browser url) you may want to return something like: `<h1>Plant not found</h1>` or something similar.
Else destructuring the `plant` variable in line 19 will fail.